implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migrator.rs

```diff
@@ -102,6 +83,36 @@ impl Migrator {
             ));
         }
 
+        let migrations = {
+            let sql_iterator = self.sql_migrator.migrations.iter().filter_map(|m| {
+                if version_interval.version_is_inside(m.version) {
+                    Some(NextMigration::Sql(m))
+                } else {
+                    None
+                }
+            });
+
+            // When running a test for a rust migration, we start the test running all the
+            // migrations right before the rust migration we want to test. This includes the sql
+            // migration that exists for the same version. We modify `version_interval` to ensure
+            // that only the sql migration is run, so that we can then populate the database with
+            // test data, and finally run the rust migration that is being tested.
+            #[cfg(test)]
```

Comment:
The code that behaves differently depending if `test` is enabled is misleading. Twice misleading when you try to test it from another crate and notice that `test` isn't propagated

Reply:
A much better approach would be to either have more clear public API to invoke desired behaviour, or to create a dedicated function (with clear naming) that is only available with `test` flag

Reply:
I'm going to try to refactor the `VersionInterfal` to include this flag. Something like:

```
struct VersionRange {
    range: core::ops::Range<Version>,
    #[cfg(test)]
    skip_last_rust_migration: bool,
}
```

Reply:
I've moved the logic of skipping the last rust migration up to the higher level, at the `Migrator`'s test function [here](https://github.com/build-trust/ockam/blob/01f455a066689606d8039c4394cdb2e4a5803a16/implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migrator.rs#L195).